### DISCUSSION TITLE
What kind of column name should I pass to the OrderBy annotation ?

### DISCUSSION THREAD

**Discussion by NoiseByNorthwest** [doctrine/doctrine2#8727] [Support Questions]
N.B. : I use `UnderscoreNamingStrategy` as naming strategy.

If I specify the mapping name like this:
```
@ORM\OrderBy({"createdAt" = "ASC", "id" = "ASC"})
```

Then the `doctrine:schema:validate` is OK but I got, when querying, this error: 

```
...
Undefined column: 7 ERROR:  column "createdAt" does not exist
...
ORDER BY "createdAt...
...
HINT:  Perhaps you meant to reference the column "XXX.created_at"
```

And If I specify the physical name like this:

```
@ORM\OrderBy({"created_at" = "ASC", "id" = "ASC"})
```

Then I got no error on query side, but the `doctrine:schema:validate` returns this error:

```
 [FAIL] The entity-class XXXX mapping is invalid:
 * The association XXXX is ordered by a foreign field created_at that is not a field on the target entity XXXX.

```

**Comment by greg0ire** [comment]
What you call a physical name is a column name. What you call a mapping name is a field name.
[In the docs, it states that you should use field names](https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/annotations-reference.html#orderby).

That means that the issue you are having must be about the generated SQL. Is there anything special about `createdAt`? Have you experienced the same issue with other fields that would be less special, but still have several words in them?

**Comment by NoiseByNorthwest** [comment]
@greg0ire It was a bug on my side, in a component which generates a native query.
